### PR TITLE
Clear old upload path from documents-api

### PR DESCRIPTION
**Pull request by bella-cockrell**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/DOC-675

### *What is the problem we're trying to solve*

Now that we have implemented the new upload path using a presigned url from S3, we need to clean up the code and remove the old path that uses the base64 encoding. 

### *What changes have we introduced*

In this commit:

- Removed old upload document path
- Amended the CreateClaimAndDocument controller and use case to return the S3 Upload Policy. 
- Update the tests
- Add comments to flag areas where the old download path needs to be removed.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### *Follow up actions after merging PR*

Are there any next steps that need to addressed after merging this PR? Add them here.

### CODE REVIEW

**Review by bogdan-zaharia — COMMENTED** [review]
Looks good, left a few comments

File: DocumentsApi/V1/Boundary/Response/CreateClaimAndS3UploadPolicyResponse.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using System;
+using DocumentsApi.V1.Domain;
+
+namespace DocumentsApi.V1.Boundary.Response
+{
+    public class CreateClaimAndS3UploadPolicyResponse
+    {
+        public Guid ClaimId { get; set; }
+        public DateTime CreatedAt { get; set; }
+        public Document Document { get; set; }
```
[the remark this answers is not in the document]
Comment (bella-cockrell) [reply DocumentsApi/V1/Boundary/Response/CreateClaimAndS3UploadPolicyResponse.cs:10]: 👍

File: DocumentsApi/V1/Factories/ResponseFactory.cs [hunk 2]
```diff
@@ -47,5 +48,21 @@ public static ClaimAndDocumentResponse ToClaimAndDocumentResponse(this ClaimResp
...
             };
         }
+
+        public static CreateClaimAndS3UploadPolicyResponse ToClaimAndS3PolicyResponse(this Claim claim, S3UploadPolicy s3UploadPolicy)
+        {
+            return new CreateClaimAndS3UploadPolicyResponse
+            {
+                ApiCreatedBy = claim.ApiCreatedBy,
+                CreatedAt = claim.CreatedAt,
+                Document = claim.Document,
```
[the remark this answers is not in the document]
Comment (bella-cockrell) [reply DocumentsApi/V1/Factories/ResponseFactory.cs:58]: 👍

File: DocumentsApi/V1/Factories/ResponseFactory.cs [hunk 3]
```diff
@@ -47,5 +48,21 @@ public static ClaimAndDocumentResponse ToClaimAndDocumentResponse(this ClaimResp
                 Base64Document = base64Document
             };
         }
+
+        public static CreateClaimAndS3UploadPolicyResponse ToClaimAndS3PolicyResponse(this Claim claim, S3UploadPolicy s3UploadPolicy)
```
[the remark this answers is not in the document]
Comment (bella-cockrell) [reply DocumentsApi/V1/Factories/ResponseFactory.cs:52]: 👍

File: DocumentsApi/V1/UseCase/CreateClaimAndS3UploadPolicyUseCase.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            {
+                _logger.LogError("VALIDATION: {0}", validation.Errors.First().ErrorMessage);
+                throw new BadRequestException(validation);
+            }
+
+            var claim = BuildClaimRequest(request);
+            var createdClaim = _documentsGateway.CreateClaim(claim);
+            var s3UploadPolicy = await _s3Gateway.GenerateUploadPolicy(createdClaim.Document).ConfigureAwait(true);
+
+            return createdClaim.ToClaimAndS3PolicyResponse(s3UploadPolicy);
```
[the remark this answers is not in the document]
Comment (bella-cockrell) [reply DocumentsApi/V1/UseCase/CreateClaimAndS3UploadPolicyUseCase.cs:44]: 👍

**Review by bogdan-zaharia — APPROVED** [review]
LGTM

**Pull request merged**